src/ts/ssh-tcp/services/portForwardingService.ts

```diff
@@ -370,14 +391,16 @@ export class PortForwardingService extends SshService {
 		}
 
 		remotePort = streamer.remotePort;
-		const remoteEndPoint = `${remoteIPAddress}:${remotePort}`;
-		this.remoteConnectors.set(remoteEndPoint, streamer);
+
+		// The remote port is the port sent in the message to the other side,
+		// so the connector is indexed on that port number. (There is no local port anyway.)
+		this.remoteConnectors.set(remotePort, streamer);
 
 		const forwardedPort = new ForwardedPort(null, remotePort, false);
 		this.localForwardedPorts.addPort(forwardedPort);
 		streamer.onDisposed(() => {
 			this.localForwardedPorts.removePort(forwardedPort);
-			this.remoteConnectors.delete(remoteEndPoint);
+			this.remoteConnectors.delete(remotePort);
```

Comment:
Same here: should we check whether `remotedConnectors[remotePort]` is still `streamer`?

Reply:
Not necessary because processing of received messages is serialized.